### ISSUE TITLE
clang 12 with flags + ppc64le TAP parsing error: Too few tests run

### ISSUE THREAD

**Issue by junaruga** [simd-everywhere/simde#745]
On the master commit: `22609d4661340098a17bddb9b3a7b8b18680079a` that is really not the latest commit. 
I ran the tests to check if the #741 was fixed on the commit with native ppc64le and clang 12.0.0 on Fedora building system. Then I got a new error. Could you take a look at it? Thanks.

```
TAP parsing error: Too few tests run (expected 408, got 234)
```

```
 487/1090 x86/svml/emul/c                     ERROR           0.82s   killed by signal 11 SIGSEGV
```

Here is the log.

```
=== 2.3. tests on clang with flags macro ===
+ BUILD_DIR=build/clang-flags-macro
+ CC=clang
+ CXX=clang++
+ CFLAGS='-O2 -flto -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS --config /usr/lib/rpm/redhat/redhat-hardened-clang.cfg -fstack-protector-strong   -m64 -mcpu=power8 -mtune=power8 -fasynchronous-unwind-tables -fstack-clash-protection'
+ CXXFLAGS='-O2 -flto -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS --config /usr/lib/rpm/redhat/redhat-hardened-clang.cfg -fstack-protector-strong   -m64 -mcpu=power8 -mtune=power8 -fasynchronous-unwind-tables -fstack-clash-protection'
+ _setup
+ meson setup build/clang-flags-macro
The Meson build system
Version: 0.57.1
Source dir: /builddir/build/BUILD/simde-22609d4661340098a17bddb9b3a7b8b18680079a
Build dir: /builddir/build/BUILD/simde-22609d4661340098a17bddb9b3a7b8b18680079a/build/clang-flags-macro
Build type: native build
Project name: SIMDe
Project version: 0.7.3
C compiler for the host machine: clang (clang 12.0.0 "clang version 12.0.0 (Fedora 12.0.0-0.7.rc3.fc35)")
C linker for the host machine: clang ld.bfd 2.36.1-8
C++ compiler for the host machine: clang++ (clang 12.0.0 "clang version 12.0.0 (Fedora 12.0.0-0.7.rc3.fc35)")
C++ linker for the host machine: clang++ ld.bfd 2.36.1-8
Host machine cpu family: ppc64
Host machine cpu: ppc64le
Library m found: YES
Compiler for C supports arguments -fopenmp-simd: YES 
Compiler for C++ supports arguments -fopenmp-simd: YES 
Compiler for C supports arguments -Wpsabi: YES 
Compiler for C++ supports arguments -Wpsabi: YES 
Library m found: YES
Build targets in project: 1090
Found ninja-1.10.2 at /usr/bin/ninja
+ _build
+ rm -f build.log
+ _time ninja -C build/clang-flags-macro -v -j 8
+ head -4 build.log
+ /usr/bin/time -f '=> [%E]' ninja -C build/clang-flags-macro -v -j 8
ninja: Entering directory `build/clang-flags-macro'
[1/2180] clang -Itest/common/common-c.p -Itest/common -I../../test/common -I. -I../.. -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -O2 -flto -fexceptions -g -grecord-gcc-switches -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS --config /usr/lib/rpm/redhat/redhat-hardened-clang.cfg -fstack-protector-strong -m64 -mcpu=power8 -mtune=power8 -fasynchronous-unwind-tables -fstack-clash-protection -fopenmp-simd -DSIMDE_ENABLE_OPENMP -Wno-psabi -DSIMDE_TEST_BARE -MD -MQ test/common/common-c.p/common.c.o -MF test/common/common-c.p/common.c.o.d -o test/common/common-c.p/common.c.o -c ../../test/common/common.c
[2/2180] clang++ -Itest/common/common-cpp.p -Itest/common -I../../test/common -I. -I../.. -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -g -O2 -flto -fexceptions -g -grecord-gcc-switches -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS --config /usr/lib/rpm/redhat/redhat-hardened-clang.cfg -fstack-protector-strong -m64 -mcpu=power8 -mtune=power8 -fasynchronous-unwind-tables -fstack-clash-protection -fopenmp-simd -DSIMDE_ENABLE_OPENMP -Wno-psabi -DSIMDE_TEST_BARE -MD -MQ test/common/common-cpp.p/meson-generated_.._common.cpp.o -MF test/common/common-cpp.p/meson-generated_.._common.cpp.o.d -o test/common/common-cpp.p/meson-generated_.._common.cpp.o -c test/common/common.cpp
+ tail -3 build.log
6 warnings generated.
[2180/2180] clang++  -o test/wasm/xor-native-cpp test/wasm/xor-native-cpp.p/meson-generated_.._xor.cpp.o -Wl,--as-needed -Wl,--no-undefined -O2 -flto -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS --config /usr/lib/rpm/redhat/redhat-hardened-clang.cfg -fstack-protector-strong -m64 -mcpu=power8 -mtune=power8 -fasynchronous-unwind-tables -fstack-clash-protection -Wl,--start-group -lm -Wl,--end-group
=> [5:49.22]
+ _test
+ _time meson test -C build/clang-flags-macro -q --no-rebuild --print-errorlogs
+ /usr/bin/time -f '=> [%E]' meson test -C build/clang-flags-macro -q --no-rebuild --print-errorlogs
 487/1090 x86/svml/emul/c                     ERROR           0.82s   killed by signal 11 SIGSEGV
>>> MALLOC_PERTURB_=231 /builddir/build/BUILD/simde-22609d4661340098a17bddb9b3a7b8b18680079a/build/clang-flags-macro/test/x86/svml-emul-c
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
TAP parsing error: Too few tests run (expected 408, got 234)
(test program exited with status code -11)
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Summary of Failures:
 487/1090 x86/svml/emul/c                     ERROR           0.82s   killed by signal 11 SIGSEGV
Ok:                 1089
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
Full log written to /builddir/build/BUILD/simde-22609d4661340098a17bddb9b3a7b8b18680079a/build/clang-flags-macro/meson-logs/testlog.txt
Command exited with non-zero status 1
=> [0:06.90]
```

**Comment by nemequ** [comment]
This looks like the same issue as #741; clang 12 seems to be broken on ppc64le with `-fstack-clash-protection` :(

I've reported the issue to LLVM: https://bugs.llvm.org/show_bug.cgi?id=49903.  Hopefully someone there who is much more comfortable with PPC can take a look.

**Comment by junaruga** [comment]
OK, thanks!

**Comment by junaruga** [comment]
Seeing the above llvm ticket, The following commit fixes this issue. But it seems it is not released yet.
https://github.com/llvm/llvm-project/commit/bf58600badb1138a501ad81b07298207a7a64b2a

**Comment by junaruga** [comment]
Ah no, it was backported to the release/12.x branch. The released clang 12.0.1 has this commit. :)
https://github.com/llvm/llvm-project/commit/04a68288ded459c7e76135a9ee4b7e9d4bf4cdc2

**Comment by junaruga** [comment]
I was able to confirm this error doesn't happen with the simde master commit `22609d4661340098a17bddb9b3a7b8b18680079a` on Fedora rawhide clang 12.0.1 any more. I would close this ticket. Thanks.
